### ISSUE TITLE
Possible misbehavior when passing Lists to function on "Wait Until Keyword Succeeds"

### ISSUE THREAD

**Issue by hernandik** [robotframework/robotframework#4391]
I identify that the follow example below won't succeed when executing the last 2 step of the example test:
I tested on robotframework==4.1.2 and robotframework==5.0.1
Summarizing, when passing list to a Wait Until Keyword Succeeds func parameter, it loses their 'meaning' and the FOR loop is unable to access their elements. Is that a expected behavior or is it an issue?

Looking into the source code for the function I notice that Run keyword is called, and it presents the same behavior.
I edited the issue and added a case for run keyword

```robot
*** Test Cases ***
Example should pass
    @{list1} =    Create List    first
    caller ${list1}
    caller2   ${list1}

    @{list2} =    Create List    first   second
    caller ${list2}
    caller2   ${list2}
    
    indirect call   ${list1}
    indirect call   ${list2}

    Run Keyword    caller2   ${list1}
    Run Keyword    caller ${list2}    # FAIL :(

    wait test2  ${list2}   # This way it works
    wait test1   ${list1}   # THIS way it FAIL
    wait test1   ${list2}

*** Keywords ***
wait test1
    [Arguments]  ${list}
    Wait Until Keyword Succeeds     1 times     1ms     caller ${list}

wait test2
    [Arguments]  ${list}
    Wait Until Keyword Succeeds     1 times     1ms     caller2  ${list}


indirect call
    [Arguments]  ${list}
    caller ${list}

caller ${list}
    FOR    ${file}    IN    @{list}
        LOG   ${file}
    END

caller2
    [Arguments]  ${list}
    FOR    ${file}    IN    @{list}
        LOG   ${file}
    END
```

**Comment by pekkaklarck** [comment]
This seems to be a variation of #1595. Can you check that?

**Comment by hernandik** [comment]
Yes, is the same behavior of https://github.com/robotframework/robotframework/issues/1595

**Comment by pekkaklarck** [comment]
Closing as a duplicate. Good news is that #1595 will be included in RF 5.1!
